include/libtorrent/alert_types.hpp

```diff
@@ -2864,6 +2865,9 @@ TORRENT_VERSION_NAMESPACE_2
 
 		std::string message() const override;
 
+		// id of the node the request was sent to (and this response was received from)
+		sha1_hash node_id;
```

Comment:
this change cannot be made in `RC_1_2` since it changes the ABI. It has to go in `master`